src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -861,17 +861,25 @@ private static License DeserializeLicense(Stream stream)
             _serverStore.NotificationCenter.Add(alert);
         }
 
-        private static void SetAffinity(Process process, int cores)
+        private void SetAffinity(Process process, int cores)
         {
             if (ProcessorInfo.ProcessorCount < cores)
                 return;
 
             try
             {
-                var bitMask = 1;
-                for (var i = 1; i <= cores; i++)
+                var bitMask = 1L;
+                var processAffinityMask = _serverStore.Configuration.Server.ProcessAffinityMask;
+                if (processAffinityMask != null && NumberOfSetBits(processAffinityMask.Value) <= cores)
```

Comment:
Use the first N bits from the number, instead

Reply:
done